src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,180 @@
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.toObject(MedicationEntity.class);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments;
+        try {
+            medicationDocuments = future.get().getDocuments();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            externalList.add(getEachInternalList(medicationDocument));
+        }
+        return externalList;
+    }
+
+    private Map<List<String>, Object> getEachInternalList(QueryDocumentSnapshot medicationDocument){
+        List<String> pks = new ArrayList<>();
+        String aux;
+        Map<List<String>, Object> internalList = new HashMap<>();
+        aux = medicationDocument.getId();
+        pks.add(pkToDate(aux));
+        pks.add(pkToName(aux));
+        internalList.put(Collections.singletonList(DATENAME), pks);
+        pks.clear();
+        internalList.put(Collections.singletonList(BODY),
+                medicationDocument.toObject(MedicationEntity.class));
+        return internalList;
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationsBetween(String owner, String petName,
+                                                                    String initialDate, String finalDate)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments;
+                try {
+                    medicationDocuments = future.get().getDocuments();
+                }  catch (InterruptedException | ExecutionException e) {
+                    e.printStackTrace();
+                    throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+                }
+                for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            if (initialDate.compareTo(pkToDate(medicationDocument.getId())) < 0 &&
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['&&' should be on a new line.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)